awpy/visualization/plot.py

```diff
@@ -326,3 +341,382 @@ def plot_nades(
+    )
+    if save_filepath:
+        figure.savefig(fname=save_filepath, bbox_inches="tight", dpi=400)
+        plt.close(figure)
+    elif not given_fig_ax[0]:
+        plt.show(figure)
+
+
+def plot_frame_map_control(
+    map_name: str,
+    frame: GameFrame,
+    plot_type: str = "",
+    given_fig_ax: tuple[plt.Figure, plt.Axes] | tuple[None, None] = (None, None),
+    save_filepath: str = "",
+) -> None:
+    """Visualize map control for awpy frame.
+
+    Args:
+        map_name (str): Map used position_transform call
+        frame (GameFrame): awpy frame to calculate map control for
+        plot_type (str): Determines which type of plot is created
+            (either with or without players)
+        given_fig_ax: Optional tuple containing figure and ax objects for plotting
+        save_filepath (str): Filepath to save the figure to file if required
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    player_positions = extract_player_positions(frame)
+    map_control_dict = calc_map_control_values(map_name, player_positions)
+    if plot_type == "players":
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            player_pos=player_positions,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+    else:
+        plot_map_control_snapshot(
+            map_name,
+            map_control_dict,
+            given_fig_ax=given_fig_ax,
+            save_filepath=save_filepath,
+        )
+
+
+def create_round_map_control_gif(
+    map_name: str,
+    round_data: GameRound,
+    plot_type: str = "",
+    gif_filepath: str = "./results/round_mc.gif",
+) -> None:
+    """Create gif summarizing map control for round.
+
+    Args:
+        map_name (str): Map used in plot_frame_map_control call
+        round_data (GameRound): Round whose map control will be animated.
+            Expected format that of awpy round
+        plot_type (str): Determines which type of plot is created
+            (either with or without players)
+        gif_filepath (str): Filepath to save the gif to file
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+                    If roundData not of expected
+                    awpy frame format
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+    if "frames" not in round_data:
+        msg = "round_data argument not of expected awpy round format."
+        raise ValueError(msg)
+    if "tmp" not in os.listdir():
+        error_string = (
+            "Frame visualizations cannot be saved to the"
+            "./tmp folder - it does not exist"
+        )
+        raise ValueError(error_string)
+
```

Comment:
You should probably just create this if it doesnt exist instead of throwing. You should also harmonize it and use the same tmp folder name as the other plot function.

Ideally what we should do it have a separate function that checks for the folder existence and throws if it already exists. if not it creates it. And at the end of the plotting function we delete the folder again. And we want to throw if it already exists because we dont want to overwrite anything.

In that case we should make the folder a module level private global variable

`_TMP_FOLDER = csgo_tmp`.

Reply:
Yeah that makes sense. Added _TMP_FOLDER to init in the vis module. Comment in general is addressed in next commit.